quark/subproject.py

```diff
@@ -348,26 +348,25 @@ def checkout(self):
                     opts = [ local_branch ]
                 fork(['git', 'checkout'] + opts + ['--'])
 
-    def update(self, clean=False):
+    def update(self, clean=False, fix_remotes=False):
         def actualUpdate():
             with cd(self.directory):
                 try:
                     current_origin = log_check_output(['git', 'config', '--get', 'remote.origin.url']).strip().decode('utf-8')
                 except CalledProcessError:
                     current_origin = None
                 if current_origin != self.url.geturl():
-                    # For now we just throw a fit; it shouldn't happen often,
-                    # and in this case there's no "right" answer - the repo may
-                    # have just moved, or we may be dealing with a completely
-                    # unrelated repo.
-                    # In future, it would be nice to be a bit more interactive
-                    raise QuarkError("""
-
-Directory '%s' is a git repository,
-but its remote 'origin' (%r)
-does not match what we expect (%r).
-
-Please either remove the local clone, or fix its remote.""" % (self.directory, current_origin, self.url.geturl()))
+                    if fix_remotes:
+                        logger.info("Fixing incorrect remote in %s directory (%s -> %s)" % (self.directory, current_origin, self.url.geturl()))
+                        fork(['git', 'remote', 'set-url', 'origin', self.url.geturl()])
+                    else:
+                        raise QuarkError("""
+
```

Comment:
Is there a reason why now this and the remaining lines in the string are indented?